c10/core/MemoryFormat.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <c10/core/Backend.h>
+#include <c10/util/Exception.h>
+
+#include <iostream>
+
+namespace c10 {
+enum class MemoryFormat : int8_t { Any, Preserve, Contiguous, ChannelsFirst };
```

Comment:
I know you are aware of this, but I want to explicitly point out that if we are going to go with this enum, we MUST NOT ever return MemoryFormat from operations which interrogate the state of a tensor. We made this mistake with Device, where we mixed up "device specification" (any CUDA device - OK) with "actual device" (any CUDA device - not OK; the index must be materialized). This is a *very* good thing to put in the comments (which I assume are eventually going to show up in this PR, since it's still WIP ;)

Reply:
I havn't planned any interrogation operations yet. But just to be clear, we are talking about external API or internal as well?

Reply:
And just to be on safe side, right now I want to have Object I can pass from python to cpp. And rewrite generator to support it. I'm not absolutely sure about class at::MemoryFormat yet.

Reply:
Both internal and external. I know you haven't planned any interrogation ops; the point is to make sure someone else doesn't decide to add one ;)